lib/src/main/scala/spinal/lib/bus/regif/BusIfAdapter/BRAMBusInterface.scala

```diff
@@ -0,0 +1,43 @@
+package spinal.lib.bus.regif
+
+import spinal.core._
+import spinal.lib.bus.misc.SizeMapping
+import spinal.lib.bus.bram.BRAM
+import spinal.lib.Delay
+
+case class BRAMBusInterface(bus: BRAM, sizeMap: SizeMapping, regPre: String = "")(implicit moduleName: ClassName) extends BusIf {
+  override val withStrb: Boolean = true
+  override val wstrb: Bits = Bits(strbWidth bit)
+  override val wmask: Bits = Bits(busDataWidth bit)
+  override val wmaskn: Bits = Bits(busDataWidth bit)
+  wstrb := bus.we
+  initStrbMasks()
+
+  override val askWrite: Bool = bus.we.orR.allowPruning()
+
+  override val askRead: Bool = (!askWrite).allowPruning()
+
+  override val doWrite: Bool = (askWrite && bus.en).allowPruning()
+
+  override val doRead: Bool = (askRead && bus.en).allowPruning()
+
+  override val readData: Bits = Bits(busDataWidth bits)
+
+  bus.rddata := readData   //Delay(readData, bus.config.readLatency) init B(0),  readData already reg-out
```

Comment:
@jijingg Why did you make this change? readData is not already registered by default, see the other regif interfaces. Even if it was, this would break the case when readLatency > 1.

Reply:
@jijingg Is it okay if I remove your commit? Happy to discuss any concerns you may have.